### PR TITLE
feat: upgrade `TablePacking` and add trace lengths safeguards 

### PR DESCRIPTION
**Pull request by Nashtare**
## Summary

Apply the logic of `TablePacking` to the Public table too and add measures to handle edge cases related to trace lengths smaller than specified configuration parameters.

## Why

First change is due to performance reason, as we can "freely" change the layout of the `PublicTable` trace, similarly as how it is done for the Witness bus and the Add / Mul Tables.

Second change is twofold:
- `TablePacking` may be incompatible with a given trace length, leading to prover failures

## Changes
- [x] Allow `PublicTable` trace to have a wider layout
- [x] Add warning upon conflicting trace length / specified table packing parameter and embeds new packing parameter in the proof itself
- [x] ~Also fix an unrelated bug noticed while implementing FRI with MMCS~ -> fixed in latest `main` independently

## Tests
- [x] Unit tests added/updated
- [x] Existing tests pass
- [ ] Edge cases considered (briefly note any)

## Benchmarks / Performance (if relevant)
- [ ] Improving E2E examples on other WIP branches

### CODE REVIEW

**Review by LindaGuiga — APPROVED** [review]
LGTM thanks!

File: circuit-prover/src/batch_stark_prover.rs [hunk 1]
```diff
@@ -3629,4 +3774,126 @@ mod tests {
...
+        let prover = BatchStarkProver::new(cfg);
+
+        let proof = prover
+            .prove_all_tables(&traces, &common, witness_multiplicities)
+            .unwrap();
+        prover.verify_all_tables(&proof, &common).unwrap();
+    }
+
+    #[test]
+    fn test_minimal_circuit_all_tables_padded() {
```
[the remark this answers is not in the document]
Comment (Nashtare) [reply circuit-prover/src/batch_stark_prover.rs]: Hmm fair I'll just remove it!

File: circuit-prover/src/batch_stark_prover.rs [hunk 2]
```diff
@@ -3629,4 +3774,126 @@ mod tests {
...
+        let mut builder = CircuitBuilder::<BabyBear>::new();
+        let cfg = config::baby_bear().build();
+
+        let x = builder.add_public_input();
+        let y = builder.add_public_input();
+        let expected = builder.add_public_input();
+
+        // Only multiplication, no addition
+        let xy = builder.mul(x, y);
+        let diff = builder.sub(xy, expected); // sub uses add internally but with negation
```
[the remark this answers is not in the document]
Comment (Nashtare) [reply circuit-prover/src/batch_stark_prover.rs]: I messed up rebasing when changing the tests, will just remove the Sub!
